ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+correspond to STORED definitions in the ECL.  Our example contains a single string "searchName", so the
+workunit contains a single input parameter::
+
+ <Variables>
+  <Variable name="searchname">
+   <SchemaRaw xsi:type="SOAP-ENC:base64">
+    searchname&#xe000;&#xe004;&#241;&#255;&#255;&#255;&#xe001;ascii&#xe000;&#xe001;ascii&#xe000;&#xe000;&#xe018;&#xe000;&#xe000;&#xe000;&#xe000;   </SchemaRaw>
+  </Variable>
+ </Variables>
+
+Results and Schemas
+-------------------
+
+The workunit xml also contains details of each result that the query generates, including a serialized description of the output record format::
+
+ <Results>
+  <Result isScalar="0"
+          name="Result 1"
+          recordSizeEntry="mf1"
+          rowLimit="-1"
+          sequence="0">
+   <SchemaRaw xsi:type="SOAP-ENC:base64">
+    name&#xe000;&#xe004;(&#xe000;&#xe000;&#xe000;&#xe001;ascii&#xe000;&#xe001;ascii&#xe000;address&#xe000;&#xe004;P&#xe000;&#xe000;&#xe000;&#xe001;ascii&#xe000;&#xe001;ascii&#xe000;&#xe000;&#xe018;%&#xe000;&#xe000;&#xe000;{ string40 name, string80 address };&#10;   </SchemaRaw>
+  </Result>
+  <Result name="Result 2" sequence="1">
+   <SchemaRaw xsi:type="SOAP-ENC:base64">
+    Result_2&#xe000;&#xe004;&#241;&#255;&#255;&#255;&#xe001;ascii&#xe000;&#xe001;ascii&#xe000;&#xe000;&#xe018;&#xe000;&#xe000;&#xe000;&#xe000;   </SchemaRaw>
+  </Result>
+ </Results>
+ 
+in our example there are two - the dataset of results and the text string "Done!".  The results of a query are
+associated with the workunit.  (They are currently saved in dali, but this may change in version 6.0.)
+
+Timings and statistics
+----------------------
+Any timings generated when compiling the query are included in the workunit dll::
+
+     <Statistics>
+      <Statistic c="eclcc"
+                 count="1"
+                 creator="eclcc"
+                 kind="SizePeakMemory"
+                 s="compile"
+                 scope="compile"
+                 ts="1428933081084000"
+                 unit="sz"
+                 value="27885568"/>
+     </Statistics>
+
+Other statistics and timings created when running the query are stored in the runtime copy of the workunit.
+(Statistics for graph elements are stored in a different format from global statistics, but the IWorkUnit interface
+ensures the implementation details are hidden.)
+
+Manifests
+---------
+
+It is possible to include other user-defined resources in the workunit dll - e.g., web pages, or dashboard layouts.
+I have to confess I don't understand them... ??Tony please provide some more information....!
+
+
+Stages of execution
+*******************
+Once a workunit has been compiled to a dll it is ready to be executed.  Execution can be triggered in different ways:
+
+* A query is submitted to a queue.  The query is compiled and eclccserver adds a workunit to a queue ready for execution.
+* A query is submitted and published.  Another request is then submitted to execute this previously compiled query.
```

Comment:
Submitted to what, the queue again? What is meant by 'published'? This would be a perfect chance to explain this. I, myself, don't know what it means to 'publish' a query. i.e. to whom, to where, how etc?